pkg/internal/otelsdk/sdk_inject.go

```diff
@@ -0,0 +1,296 @@
+package otelsdk
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"log/slog"
+	"maps"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/grafana/jvmtools/jvm"
+
+	"github.com/grafana/beyla/v2/pkg/beyla"
+	"github.com/grafana/beyla/v2/pkg/export/otel"
+	"github.com/grafana/beyla/v2/pkg/internal/ebpf"
+	ebpfcommon "github.com/grafana/beyla/v2/pkg/internal/ebpf/common"
+	"github.com/grafana/beyla/v2/pkg/internal/svc"
+)
+
+const OTelJDKAgent = "opentelemetry-javaagent.jar"
```

Comment:
As a future task, I'd set this value configurable so the user can provide its own version of the java-agent (e.g. a patched version, or an updated version to temporarily fix a discovered vulnerability).